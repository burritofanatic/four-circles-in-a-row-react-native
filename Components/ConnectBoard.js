import React, {Component} from 'react';
import {View, Text, StyleSheet, Dimensions} from 'react-native';
import * as Animatable from 'react-native-animatable';
import ColumnButton from "../Components/ColumnButton";
import * as Constants from '../Constants';

const width = Dimensions.get('window').width;
const height = Dimensions.get('window').height;

const NUMBER_OF_ROWS = Constants.NUMBER_OF_ROWS;
const WIDTH_OF_BOARD_TO_SCREEN = Constants.WIDTH_OF_BOARD_TO_SCREEN;

class ConnectBoard extends Component {

  onColumnClicked = (column) =>  {
    this.props.onColumnClicked(column);
  }

  onColumnPressedIn = (column) =>  {
    this.setState({ activeColumn: column });
  }

  returnButtonForRow = (value, row, col) => {
    /*This function determines the color of the botton to populate on the position based on the existing value from the 2D array. */

    switch (value) {
      case 1:
        return <View key={row, col} style={styles.buttonContainerOne} />;
      case 2:
        return <View key={row, col} style={styles.buttonContainerTwo} />;
      default:
        return <View key={row, col} style={styles.buttonContainerEmpty} />;
    }
  }

  returnOffsetForIndex = (index) => {
    if (index == 0) {
      return 0;
    } else {
      return ((width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS) * index;
    }
  }

  returnRows = (board) => {
    return board.map((row, rowIndex) => {
      return <View key={row + rowIndex} style={styles.row}>
        {row.map((value, colIndex) => {
          return this.returnButtonForRow(value, 0, colIndex)
        })
        }
      </View>
    })
  }

  render() {
    let board = this.props.board;
    let gameOver = this.props.gameOver;
    return (
      <Animatable.View animation='zoomIn' style={styles.board}>
        {
          board[0].map((value, colIndex) => {
            return <ColumnButton column={colIndex}
                      isGameOver={gameOver}
                      onColumnClicked={this.onColumnClicked}
                      key={"column-button-0-" + colIndex}
                      offset={{left: this.returnOffsetForIndex(colIndex)}}
                      onColumnPressedIn={this.onColumnPressedIn}/>
          })
        }

        { this.returnRows(board) }
      </Animatable.View>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  board: {
    flex: 1,
    backgroundColor: 'skyblue',
    borderRadius: 25,
    zIndex: 2,
    alignItems: 'center',
    justifyContent: 'center'
  },
  buttonContainerOne: {
    borderRadius: 25,
    width: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    height: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    backgroundColor: 'red'
  },
  buttonContainerTwo: {
    borderRadius: 25,
    width: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    height: (width * 0.8) / NUMBER_OF_ROWS,
    backgroundColor: 'black'
  },
  buttonContainerEmpty: {
    borderRadius: 25,
    width: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    height: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    backgroundColor: 'white'
  },
})

export default ConnectBoard;
