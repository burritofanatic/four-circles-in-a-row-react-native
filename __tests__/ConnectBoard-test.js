import React from 'react';
import ConnectBoard from '../Components/ConnectBoard';
import renderer from 'react-test-renderer';

it('renders correctly', () => {

  const board = [[0,0,0,0,0,0,0],
          [0,0,0,0,0,0,0],
          [0,0,0,0,0,0,0],
          [0,0,0,0,0,0,0],
          [0,0,0,0,0,0,0],
          [0,0,0,0,0,0,0]];

  const tree = renderer
        .create(<ConnectBoard board={board} gameOver={true} />)
        .toJSON();
  expect(tree).toMatchSnapshot();
})
