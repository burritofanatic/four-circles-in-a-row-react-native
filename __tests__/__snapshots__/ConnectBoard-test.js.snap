// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "skyblue",
      "borderRadius": 25,
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 0.3,
        },
      ],
      "zIndex": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 0,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 85.71428571428571,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 85.71428571428571,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 171.42857142857142,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 171.42857142857142,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 257.1428571428571,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 257.1428571428571,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 342.85714285714283,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 342.85714285714283,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 428.57142857142856,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 428.57142857142856,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "left": 514.2857142857142,
        },
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 514.2857142857142,
          },
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flexDirection": "row",
            "opacity": 0.5,
            "position": "absolute",
            "top": 0,
            "width": 85.71428571428571,
            "zIndex": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 85.71428571428571,
          "width": 85.71428571428571,
        }
      }
    />
  </View>
</View>
`;
